vterm-module-make.el

```diff
@@ -2,22 +2,44 @@
 
 (require 'files)
 
-(defvar vterm-install-buffer-name " *Install vterm"
+(defvar vterm-install-buffer-name "*Install vterm*"
   "Name of the buffer used for compiling vterm-module.")
 
+(defcustom vterm-compile-with-system-libvterm nil
+  "Compile vterm with the system libvterm instead of downloading
+and a local copy from GitHub."
+  :type  'boolean
+  :group 'vterm)
+
 ;;;###autoload
 (defun vterm-module-compile ()
-  "This function compiles the vterm-module."
+  "This function compiles the vterm-module.
+
+If the variable vterm-compile-with-system-libvterm is set to t,
+the system libvterm is used."
   (interactive)
-  (let ((default-directory
-          (file-name-directory (file-truename (locate-library "vterm"))))
-        (make-commands
-         "mkdir -p build; \
-          cd build; \
-          cmake \
-            -DCMAKE_BUILD_TYPE=RelWithDebInfo \
-            ..; \
-          make"))
+  (let* ((vterm-directory
+           (file-name-directory (file-truename (locate-library "vterm"))))
+         ;; Ask the user if they want to compile with the version of libvterm
+         ;; on the system. It has to be available! (We don't perform checks)
+         (compile-with-system-libvterm
+          (if (or
+               vterm-compile-with-system-libvterm
+               (y-or-n-p
+                "Do you want to use the system libvterm? (It has to be installed separately)"))
+              (progn
+                (setq vterm-compile-with-system-libvterm t) ; If the user says "yes",
+                                                            ; we save their preference
+               "yes")
+              "no"))
+         (make-commands
+          (concat
+           "cd " vterm-directory ";"
```

Comment:
Here you definitely need `shell-quote-argument`, in case Emacs is installed into a directory whose name contains special shell chars like spaces or semicolons.